// @gorhom/bottom-sheet/lib/commonjs/components/bottomSheet/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DEFAULT_KEYBOARD_BLUR_BEHAVIOR = exports.DEFAULT_KEYBOARD_BEHAVIOR = exports.DEFAULT_ANIMATE_ON_MOUNT = exports.DEFAULT_ENABLE_FLASH_SCROLLABLE_INDICATOR_ON_EXPAND = exports.DEFAULT_ENABLE_OVER_DRAG = exports.DEFAULT_ENABLE_HANDLE_PANNING_GESTURE = exports.DEFAULT_ENABLE_CONTENT_PANNING_GESTURE = exports.DEFAULT_OVER_DRAG_RESISTANCE_FACTOR = exports.DEFAULT_HANDLE_HEIGHT = exports.DEFAULT_ANIMATION_DURATION = exports.DEFAULT_ANIMATION_EASING = void 0;

var _reactNativeReanimated = require("react-native-reanimated");

var _constants = require("../../constants");

var _easingExp = require("../../utilities/easingExp");

const DEFAULT_ANIMATION_EASING = _reactNativeReanimated.Easing.out(_easingExp.exp);

exports.DEFAULT_ANIMATION_EASING = DEFAULT_ANIMATION_EASING;
const DEFAULT_ANIMATION_DURATION = 500;
exports.DEFAULT_ANIMATION_DURATION = DEFAULT_ANIMATION_DURATION;
const DEFAULT_HANDLE_HEIGHT = 24;
exports.DEFAULT_HANDLE_HEIGHT = DEFAULT_HANDLE_HEIGHT;
const DEFAULT_OVER_DRAG_RESISTANCE_FACTOR = 2.5;
exports.DEFAULT_OVER_DRAG_RESISTANCE_FACTOR = DEFAULT_OVER_DRAG_RESISTANCE_FACTOR;
const DEFAULT_ENABLE_CONTENT_PANNING_GESTURE = true;
exports.DEFAULT_ENABLE_CONTENT_PANNING_GESTURE = DEFAULT_ENABLE_CONTENT_PANNING_GESTURE;
const DEFAULT_ENABLE_HANDLE_PANNING_GESTURE = true;
exports.DEFAULT_ENABLE_HANDLE_PANNING_GESTURE = DEFAULT_ENABLE_HANDLE_PANNING_GESTURE;
const DEFAULT_ENABLE_OVER_DRAG = true;
exports.DEFAULT_ENABLE_OVER_DRAG = DEFAULT_ENABLE_OVER_DRAG;
const DEFAULT_ENABLE_FLASH_SCROLLABLE_INDICATOR_ON_EXPAND = true;
exports.DEFAULT_ENABLE_FLASH_SCROLLABLE_INDICATOR_ON_EXPAND = DEFAULT_ENABLE_FLASH_SCROLLABLE_INDICATOR_ON_EXPAND;
const DEFAULT_ANIMATE_ON_MOUNT = false;
exports.DEFAULT_ANIMATE_ON_MOUNT = DEFAULT_ANIMATE_ON_MOUNT;
const DEFAULT_KEYBOARD_BEHAVIOR = _constants.KEYBOARD_BEHAVIOR.none;
exports.DEFAULT_KEYBOARD_BEHAVIOR = DEFAULT_KEYBOARD_BEHAVIOR;
const DEFAULT_KEYBOARD_BLUR_BEHAVIOR = _constants.KEYBOARD_BLUR_BEHAVIOR.none;
exports.DEFAULT_KEYBOARD_BLUR_BEHAVIOR = DEFAULT_KEYBOARD_BLUR_BEHAVIOR;
