// @gorhom/bottom-sheet/lib/commonjs/hooks/useKeyboard.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useKeyboard = void 0;

var _react = require("react");

var _reactNative = require("react-native");

var _reactNativeReanimated = require("react-native-reanimated");

var _constants = require("../constants");

const KEYBOARD_EVENT_MAPPER = {
  KEYBOARD_SHOW: _reactNative.Platform.select({
    ios: 'keyboardWillShow',
    android: 'keyboardDidShow',
    default: ''
  }),
  KEYBOARD_HIDE: _reactNative.Platform.select({
    ios: 'keyboardWillHide',
    android: 'keyboardDidHide',
    default: ''
  })
};

const useKeyboard = () => {
  //#region variables
  const shouldHandleKeyboardEvents = (0, _reactNativeReanimated.useSharedValue)(false);
  const keyboardState = (0, _reactNativeReanimated.useSharedValue)(_constants.KEYBOARD_STATE.UNDETERMINED);
  const keyboardHeight = (0, _reactNativeReanimated.useSharedValue)(0);
  const keyboardAnimationEasing = (0, _reactNativeReanimated.useSharedValue)('keyboard');
  const keyboardAnimationDuration = (0, _reactNativeReanimated.useSharedValue)(500); //#endregion
  //#region worklets

  const handleKeyboardEvent = (0, _reactNativeReanimated.useWorkletCallback)((state, height, duration, easing) => {
    if (state === _constants.KEYBOARD_STATE.SHOWN && !shouldHandleKeyboardEvents.value) {
      return;
    }

    keyboardState.value = state;
    keyboardHeight.value = state === _constants.KEYBOARD_STATE.SHOWN ? height : height === 0 ? keyboardHeight.value : height;
    keyboardAnimationDuration.value = duration;
    keyboardAnimationEasing.value = easing;
  }, []); //#endregion
  //#region effects

  (0, _react.useEffect)(() => {
    const handleOnKeyboardShow = event => {
      (0, _reactNativeReanimated.runOnUI)(handleKeyboardEvent)(_constants.KEYBOARD_STATE.SHOWN, event.endCoordinates.height, event.duration, event.easing);
    };

    const handleOnKeyboardHide = event => {
      (0, _reactNativeReanimated.runOnUI)(handleKeyboardEvent)(_constants.KEYBOARD_STATE.HIDDEN, event.endCoordinates.height, event.duration, event.easing);
    };

    _reactNative.Keyboard.addListener(KEYBOARD_EVENT_MAPPER.KEYBOARD_SHOW, handleOnKeyboardShow);

    _reactNative.Keyboard.addListener(KEYBOARD_EVENT_MAPPER.KEYBOARD_HIDE, handleOnKeyboardHide);

    return () => {
      _reactNative.Keyboard.removeListener(KEYBOARD_EVENT_MAPPER.KEYBOARD_SHOW, handleOnKeyboardShow);

      _reactNative.Keyboard.removeListener(KEYBOARD_EVENT_MAPPER.KEYBOARD_HIDE, handleOnKeyboardHide);
    };
  }, [handleKeyboardEvent]); //#endregion

  return {
    state: keyboardState,
    height: keyboardHeight,
    animationEasing: keyboardAnimationEasing,
    animationDuration: keyboardAnimationDuration,
    shouldHandleKeyboardEvents
  };
};

exports.useKeyboard = useKeyboard;
