// @gorhom/bottom-sheet/lib/module/components/sectionList/SectionList.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { useImperativeHandle, useEffect, useRef, forwardRef, memo } from 'react';
import { SectionList as RNSectionList } from 'react-native';
import Animated from 'react-native-reanimated';
import { NativeViewGestureHandler } from 'react-native-gesture-handler';
import BottomSheetDraggableView from '../bottomSheetDraggableView';
import { useScrollableInternal, useBottomSheetInternal } from '../../hooks';
import { styles } from './styles';
const AnimatedSectionList = Animated.createAnimatedComponent(RNSectionList);
const BottomSheetSectionListName = 'SectionList';
const BottomSheetSectionListComponent = /*#__PURE__*/forwardRef((props, ref) => {
  // props
  const {
    focusHook: useFocusHook = useEffect,
    overScrollMode = 'never',
    ...rest
  } = props; // refs

  const nativeGestureRef = useRef(null); // hooks

  const {
    scrollableRef,
    scrollableAnimatedProps,
    handleScrollEvent,
    handleSettingScrollable
  } = useScrollableInternal(BottomSheetSectionListName);
  const {
    enableContentPanningGesture
  } = useBottomSheetInternal(); // effects
  // @ts-ignore

  useImperativeHandle(ref, () => scrollableRef.current);
  useFocusHook(handleSettingScrollable); // render

  return /*#__PURE__*/React.createElement(BottomSheetDraggableView, {
    nativeGestureRef: nativeGestureRef,
    style: styles.container
  }, /*#__PURE__*/React.createElement(NativeViewGestureHandler, {
    ref: nativeGestureRef,
    enabled: enableContentPanningGesture,
    shouldCancelWhenOutside: false
  }, /*#__PURE__*/React.createElement(AnimatedSectionList, _extends({}, rest, {
    // @ts-ignore
    ref: scrollableRef,
    overScrollMode: overScrollMode,
    scrollEventThrottle: 16,
    onScrollBeginDrag: handleScrollEvent // @ts-ignore
    ,
    animatedProps: scrollableAnimatedProps
  }))));
});
const BottomSheetSectionList = /*#__PURE__*/memo(BottomSheetSectionListComponent);
BottomSheetSectionList.displayName = 'BottomSheetSectionList';
export default BottomSheetSectionList;
