// @gorhom/bottom-sheet/lib/module/components/bottomSheetContainer/BottomSheetContainer.js
import React, { memo, useCallback, useMemo } from 'react';
import { View } from 'react-native';
import { styles } from './styles';

const BottomSheetContainerComponent = ({
  shouldMeasureHeight,
  containerHeight,
  onMeasureHeight,
  children
}) => {
  //#region callbacks
  const handleOnLayout = useCallback(({
    nativeEvent: {
      layout: {
        height
      }
    }
  }) => {
    onMeasureHeight(height);
  }, [onMeasureHeight]); //#endregion
  //#region styles

  const containerStyle = useMemo(() => [styles.container, !shouldMeasureHeight && containerHeight ? {
    height: containerHeight
  } : {}], [containerHeight, shouldMeasureHeight]); //#endregion
  //#region render
  // console.log('BottomSheetContainer', 'render', shouldMeasureHeight);

  return /*#__PURE__*/React.createElement(View, {
    pointerEvents: "box-none",
    style: containerStyle,
    onLayout: shouldMeasureHeight ? handleOnLayout : undefined,
    children: children
  }); //#endregion
};

const BottomSheetContainer = /*#__PURE__*/memo(BottomSheetContainerComponent);
BottomSheetContainer.displayName = 'BottomSheetContainer';
export default BottomSheetContainer;
